### ISSUE TITLE
Not shortening URLs with certain special characters

### ISSUE THREAD

**Issue by akagno-ent** [Shagon94/short-link#1]
Hey there, would love to use this more, but I can't seem to shorten some of the most important ones to shorten!

The table:

```
| App                    | Development                      | Training                         | Staging                          | DR                              | Production                       |
| ---------------------- | -------------------------------- | -------------------------------- | -------------------------------- | ------------------------------- | -------------------------------- |
| [AppName](#FN-AppName) | [DevName](azurelink)             | [TrnName](azurelink)             | [StgName](azurelink)             | [DrName](azurelink)             | [PrdName](azurelink)             |
| [AppName](anchor)      | [DevName](https://linkwith#init) | [TrnName](https://linkwith$init) | [StgName](https://linkwith%init) | [DrName](https://linkwith&init) | [PrdName](https://linkwith@init) |
```

The output:

![image](https://github.com/Shagon94/short-link/assets/159298933/236d7da2-28a8-4761-8003-f45b1eadc8f6)

The list:

```
- [AppName](#anchor)
- [DevName](https://linkwith.com/#init) 
- [TrnName](https://linkwith.com/$init) 
- [StgName](https://linkwith.com/%init) 
- [DrName](https://linkwith.com/&init)
- [PrdName](https://linkwith.com/@init)
```

The output:

![image](https://github.com/Shagon94/short-link/assets/159298933/97bc6799-b951-4c55-a01a-92a81162bdca)

**Comment by Shagon94** [comment]
Hey, sorry, I haven't used this app in a few years, if you create any MRs I'm happy to merge it.

**Comment by Shagon94** [comment]
I believe this is the line that needs change https://github.com/Shagon94/short-link/blob/eeb1dc35f5b2290beba0d87f4cece7e1cec141ec/lib/short-link-message-dialog.js#L34

**Comment by Shagon94** [comment]
@akagno-ent now that I am near my laptop, I've double-checked and I can confirm that the regex above needs to be updated so it can match the URL formats you mentioned. You can use https://regex101.com/ to find the appropriate regex however I have no intention of maintaining this project as I switched to obsidian a few years ago.

**Comment by Shagon94** [comment]
fixed in #2
